### ISSUE TITLE
Implement interface to run JINTRAC via docker container

### ISSUE THREAD

**Issue by stefsmeets** [duqtools/duqtools#59]
Currently we are targeting to run JINTRAC via jetto on the Gateyway. A container for JINTRAC will be available shortly, which will allow us to run the tool in the cloud / locally. We need an interface to run JINTRAC via the container.

**Comment by fcasson** [comment]
We already have an interface to run JINTRAC containerised runs on PROMINENCE

https://users.euro-fusion.org/pages/data-cmg/wiki/JINTRAC_cloud.html

**Comment by v1kko** [comment]
We have the (imas enabled) docker container, so we can create a specific 'submit' type of the `jetto` system to submit it into a docker container

**Comment by fcasson** [comment]
Please clarify if you want a local submission of a container run or a remote submission.  The latter exists, if you want to add the former, I guess it belongs in jetto-tools?

For udocker, a local run is invoked like this

`udocker run -v /path/to/my/rundir:/jetto/runs/somedir $UDOCKER_NAME rjettov -x64 -S somedir build docker`

For singularity, a local run is invoked like this

`singularity exec --cleanenv -B /tmp -B /path/to/my/rundir:/jetto/runs/aaa /path/to/singularity/sim:v220922.simg rjettov -x64 -S -p0 -n1 aaa build docker`

The JINTRAC regression test script `compare_cases` has wrappers to the runscript calls which add the udocker / singularity calls.  Something similar could be added in jetto-tools for local container runs.

**Comment by v1kko** [comment]
Hi @fcasson, we want both, prominence submission via `jetto-tools` is already implemented, however it does not yet work, because we do not put the `imasdb` inside the run folder yet.

I will have a look at your suggestions of the local run, in the end this should be probably be a small function for which I agree it would belong to `jetto-tools`.

**Comment by fcasson** [comment]
Great.  Yes probably some adaptation are needed to support IMAS runs, we haven't done any of that in containers yet.

**Comment by v1kko** [comment]
This container: gitlab.eufus.psnc.pl:5050/containerization/jintrac/jintrac-imas-installer/jintrac-imas:latest seems to work quite well with IMAS runs, so I wanted to base the local running upon that container.

**Comment by fcasson** [comment]
Yes, it's very new / alpha and no one has used it yet and jetto-tools / prominence need some tweaks to be compatible, as you pointed out.

**Comment by stefsmeets** [comment]
This is no longer in scope for duqtools.

**Comment by fcasson** [comment]
That's a shame!  The IMAS container is working well now!

**Comment by stefsmeets** [comment]
I know it's working well, I use it often for testing. The reason is that our time on this project is running out and this is technically challenging. We decided to focus our energy elsewhere.

If there is demand for this or a clear use case, I'll be happy to reconsider.

**Comment by fcasson** [comment]
I guess the use case will become more relevant when users are ready to exceed the limits of the Gateway with their scans.

I assume you mean that remote submission and monitoring (e.g. coupling to promience) is technically challenging?

Running locally in the container is not that different to running locally with the continer

**Comment by stefsmeets** [comment]
We can already submit to prominence with duqtools, if that's what you mean when users are exceeding the limits of gateway. This is working (correct me if I'm wrong @v1kko), see: https://duqtools.readthedocs.io/en/latest/examples/duqtools_on_prominence/ .

We were initially looking into this to make the workflow available to users on their own PCs, and hav duqtools start jintrac via docker. See also #506

Basically, we believe we got everything covered, so this issue is no longer relevant.

**Comment by fcasson** [comment]
Okay, great - then just a misunderstanding on my part.
